conductr_cli/resolvers/uri_resolver.py

```diff
@@ -60,3 +60,49 @@ def cache_path(cache_dir, uri):
     parsed = urlparse(uri, scheme='file')
     basename = os.path.basename(parsed.path)
     return '{}/{}'.format(cache_dir, basename)
+
+
+def download_bundle(log, bundle_url, tmp_download_path):
+    log.info('Retrieving {}'.format(bundle_url))
+
+    parsed = urlparse(bundle_url, scheme='file')
+    is_http_download = parsed.scheme == 'http' or parsed.scheme == 'https'
+
+    if log.is_info_enabled() and is_http_download:
+        urlretrieve(bundle_url, tmp_download_path, reporthook=show_progress)
+    else:
+        # File based download, no need to show progress bar
+        urlretrieve(bundle_url, tmp_download_path)
+
+
+def show_progress(count, block_size, total_size):
+    downloaded_size = count * block_size
+
+    progress_bar_text = progress_bar(downloaded_size, total_size)
+    if downloaded_size < total_size:
+        print_replace(progress_bar_text)
+    else:
+        sys.stdout.write(progress_bar_text)
+        sys.stdout.write('\n')
+        sys.stdout.flush()
+
+
+def progress_bar(current_size, total_size):
+    bar_range = 50
+    percent = int(current_size * 100 / total_size)
+
+    def progress_character(number):
+        if number * (100 / bar_range) <= percent:
+            return '#'
+        else:
+            return ' '
+
+    progress = ''.join([progress_character(i) for i in range(1, bar_range)])
+    return '[{}] {}'.format(progress, '%3d%%' % percent)
+
+
+def print_replace(message):
+    flush_character = ''.join(['\b' for c in message])
```

Comment:
Interesting, why not a single `\r`?
This is message.length backspace chars right?